### ISSUE TITLE
Windows.Forms exception when opening StyleCop settings; have XQuartz installed

### ISSUE THREAD

**Issue by sfxxx** [DarkCloud14/MonoDevelop.StyleCop#14]
I'm using the MonoDevelop 4.0.1 which comes with Unity3D on OS X Yosemite. I installed MonoDevelop.StyleCop-1.0.1.8_MD4.0.mpack and XQuartz 2.77. Whenever I open the "Style Cop Settings" I get the exception below. I know nothing about XQuartz and how it's meant to help getting Window.Forms apps running. Is there anything else that has to be configured?

System.TypeInitializationException: An exception was thrown by the type initializer for System.Windows.Forms.WindowsFormsSynchronizationContext ---> System.TypeInitializationException: An exception was thrown by the type initializer for System.Windows.Forms.ThemeEngine ---> System.ArgumentException: The requested FontFamily could not be found [GDI+ status: FontFamilyNotFound]
  at System.Drawing.GDIPlus.CheckStatus (Status status) [0x00000] in <filename unknown>:0 
  at System.Drawing.FontFamily..ctor (GenericFontFamilies genericFamily) [0x00000] in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.Drawing.FontFamily:.ctor (System.Drawing.Text.GenericFontFamilies)
  at System.Drawing.FontFamily.get_GenericSansSerif () [0x00000] in <filename unknown>:0 
  at System.Drawing.Font.CreateFont (System.String familyName, Single emSize, FontStyle style, GraphicsUnit unit, Byte charSet, Boolean isVertical) [0x00000] in <filename unknown>:0 
  at System.Drawing.Font..ctor (System.String familyName, Single emSize, FontStyle style, GraphicsUnit unit, Byte gdiCharSet, Boolean gdiVerticalFont) [0x00000] in <filename unknown>:0 
  at System.Drawing.Font..ctor (System.String familyName, Single emSize, System.String systemName) [0x00000] in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.Drawing.Font:.ctor (string,single,string)
  at System.Drawing.SystemFonts.get_DefaultFont () [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.Theme..ctor () [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.ThemeWin32Classic..ctor () [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.ThemeEngine..cctor () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Windows.Forms.SystemInformation.get_MenuAccessKeysUnderlined () [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.Control..ctor () [0x00000] in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control:.ctor ()
  at System.Windows.Forms.WindowsFormsSynchronizationContext..cctor () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Windows.Forms.Control..ctor () [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.ScrollableControl..ctor () [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.ContainerControl..ctor () [0x00000] in <filename unknown>:0 
  at System.Windows.Forms.UserControl..ctor () [0x00000] in <filename unknown>:0 
  at StyleCop.CSharp.CompanyInformation..ctor () [0x00000] in <filename unknown>:0 
  at StyleCop.CSharp.CompanyInformation..ctor (StyleCop.CSharp.DocumentationRules analyzer) [0x00000] in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) StyleCop.CSharp.CompanyInformation:.ctor (StyleCop.CSharp.DocumentationRules)
  at StyleCop.CSharp.DocumentationRules.get_SettingsPages () [0x00000] in <filename unknown>:0 
  at StyleCop.StyleCopCore.GetSettingsPages (StyleCop.StyleCopCore core) [0x00000] in <filename unknown>:0

**Comment by DarkCloud14** [comment]
Did you download and install the latest stable release of XQuartz from the following link?
http://xquartz.macosforge.org/

Download the dmg, install it and at least log off and back on again after installation.
Could also be that a restart is necessary but not sure anymore.

**Comment by sfxxx** [comment]
Thanks for your reply. Yes, I have XQuartz 2.7.7 (latest release) installed and also tried a reboot.
As a workaround, I'm currently running the StyleCopSettingsEditor.exe in the add-in's package externally with the latest mono installation.

It seems there has to be one Settings.StyleCop file in the project root, next to where the cache file is put. Is there a way to get more fine-grained control using additional Settings.StyleCop files in project subfolders? Or a way to exclude certain folders (3rd party stuff) from scanning? I've tried addition settings files, but to no avail.

Here's what I did to run the editor (in case anyone googles for the same problem):

Setup:
```
# With brew and brew-cask installed
brew cask install mono-mdk
open /opt/homebrew-cask/Caskroom/mono-mdk/4.2.0/MonoFramework-MDK-4.2.0.macos10.xamarin.x86.pkg 
```
Running the editor:
`/Library/Frameworks/Mono.framework/Commands/mono "/Users/`whoami`/Library/Application Support/MonoDevelop-Unity-4.0/LocalInstall/Addins/MonoDevelop.StyleCop.1.0.1.8/StyleCopSettingsEditor.exe" <my-project-path>/Settings.StyleCop `

**Comment by DarkCloud14** [comment]
Sorry for my late response, I was a busy but I've tested it now with MD of Unity and get the same error.
I'll check if I can figure out what's going on here.

**Comment by sfxxx** [comment]
According to their roadmap (https://unity3d.com/unity/roadmap), Unity will update their MD version to 5.9 in december. There's a preview release for it (http://forum.unity3d.com/threads/monodevelop-unity-5-9-preview-release.350016/). Perhaps it's not worth fixing the old version if the next one works.

**Comment by DarkCloud14** [comment]
Hmm ok, maybe I should start working on GTK settings editor again. Thought that it wouldn't be needed but if I had it done already this wouldn't be a problem at all..

**Comment by AravSinghal** [comment]
This issue is still not resolved with v1.0.1.9 of the add-in on Xamarin Studio 6.0.1 on a Mac.

**Comment by DarkCloud14** [comment]
Just tested it and you're right, unfortunately I'm not sure if I can get it fixed at least with the original StyleCop settings. For some reason this works fine with Xamarin Studio 5.10 and addin v1.0.1.9 but with Xamarin Studio 6.0.1 the original StyleCop settings dialog crashes XS.

**Comment by chrux** [comment]
I have the same issue, StyleCop settings dialog crashes XS.

**Comment by DarkCloud14** [comment]
Ok I think the main problem is now, that Xamarin Studio 6.x is now a 64bit process and previous builds were 32bit. According to the log files the Carbon driver is not ported to 64bit and so only very few parts of System.Windows.Forms will work.

Now it's definitely the time that I get the GTK+ settings part ported.

**Comment by DarkCloud14** [comment]
I mark this as wontfix as I can't fix the problem itself.
As soon as I finish issue #4 this should also shouldn't be a problem anymore.

**Comment by DarkCloud14** [comment]
Problem will be fixed after I finally finished #4 so I close this issue.
